mcp_gateway/plugins/guardrails/dominion.py

```
@@ -0,0 +1,233 @@
+        cached = self._cache.get(server_name)
+        if cached is not None:
+            return cached
+
+        url = f"{self.api_base_url.rstrip('/')}/benchmark/{server_name}"
+        logger.debug(f"Fetching trust score from: {url}")
+
+        try:
+            req = urllib.request.Request(
+                url,
+                headers={"Accept": "application/json", "User-Agent": "mcp-gateway-dominion-plugin/1.0"},
+            )
```

Ruff (0.15.13)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)